bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandlerFactory.java

```diff
@@ -67,17 +66,13 @@ protected void activate(ComponentContext componentContext) {
     }
 
     /**
-     * Disposes the {@link BaseThingHandlerFactory}. If this method is
-     * overridden by a sub class, the implementing method must call <code>super.deactivate(componentContext)</code>
-     * first.
+     * Disposes the {@link BaseThingHandlerFactory}. If this method is overridden by a sub class, the implementing
+     * method must call <code>super.deactivate(componentContext)</code> first.
      *
      * @param componentContext
      *            component context (must not be null)
      */
     protected void deactivate(ComponentContext componentContext) {
-        for (ServiceRegistration<ThingHandler> serviceRegistration : this.thingHandlers.values()) {
```

Comment:
There is no problem. Handlers would not be kept. If a factory goes all handlers are unregistered by the ThingManager. See method `ThingManager.removeThingHandlerFactory()`.